vblagoje/auto-pr-writer
# Add docker-push workflow, rename workflow for PR generation

### Why
The pull request introduces an automation enhancement to the repository's CI/CD processes. Specifically, it adds the capability to automatically build and push Docker images to Docker Hub upon commits to the main branch, streamlining deployment workflows. Additionally, the PR includes housekeeping changes by renaming an existing workflow file for PR text generation to better reflect its functionality.

### What
- A new GitHub Actions workflow, `docker-push.yml`, has been added to build and push Docker images.
- The existing workflow, previously named `dry_run.yml`, has been renamed to `pr-text-generator.yml`.

### How can it be used
The newly added automation can be triggered by pushing code changes to the main branch. The workflow then performs the following tasks:
1. Checks out the repository.
2. Sets up Docker Buildx.
3. Logs into Docker Hub using secrets.
4. Builds the Docker image from the Dockerfile located at the root of the repository.
5. Pushes the Docker image to Docker Hub with the tag `vblagoje/auto-pr-writer:latest`.

### How did you test it
Tests should be performed to ensure the workflow executes as expected. However, testing details have not been provided. It is recommended to:
1. Push a test commit to the main branch (preferably in a test environment).
2. Check the GitHub Actions tab to confirm that the workflow runs successfully.
3. Verify that the Docker image is updated on Docker Hub.

### Notes for the reviewer
- Please double-check that the required secrets (`DOCKERHUB_USERNAME` and `DOCKERHUB_ACCESS_TOKEN`) are set up in the repository for the workflow to log in to Docker Hub successfully.
- Observe the naming conventions and paths referenced in the `docker-push.yml`, such as the Dockerfile location and image tag, ensuring they align with the project's deployment strategy.
- No modifications were made to the renamed workflow's content (`pr-text-generator.yml`), ensuring that existing functionality remains unaffected.
